Bluetooth: Mesh: Add prefix to Subnet Bridge API

HaavardRei opened the pull request:
Adds the `bt_mesh_brg_cfg` prefix to the public Subnet Bridge API, and aligns the function and callback naming with the rest of the Bluetooth Mesh API.

HaavardRei committed:
Bluetooth: Mesh: Add prefix to Subnet Bridge API

Adds the `bt_mesh_brg_cfg` prefix to the public Subnet Bridge API, and
aligns the function and callback naming with the rest of the Bluetooth
Mesh API.

Signed-off-by: Håvard Reierstad <haavard.reierstad@nordicsemi.no>

Balaklaka commented on subsys/bluetooth/mesh/brg_cfg_cli.c:
```diff
@@ -17,30 +17,29 @@ static int32_t msg_timeout;
 
-static int subnet_bridge_status(const struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
-				struct net_buf_simple *buf)
+static int bridge_status(const struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
```
This is called subnet_bridge_status in the spec. See 4.3.11.3, it is easier to link it to the spec then. This is for alle this op functions.

Balaklaka replied:
Because we are cutting the names in the header files. We can do that here too.

Balaklaka commented on include/zephyr/bluetooth/mesh/brg_cfg_cli.h:
```diff
@@ -145,17 +143,16 @@ int bt_mesh_brg_cfg_cli_subnet_bridge_get(uint16_t net_idx, uint16_t addr,
-int bt_mesh_brg_cfg_cli_subnet_bridge_set(uint16_t net_idx, uint16_t addr,
-					  enum bt_mesh_subnet_bridge_state val,
-					  enum bt_mesh_subnet_bridge_state *status);
+int bt_mesh_brg_cfg_cli_set(uint16_t net_idx, uint16_t addr, enum bt_mesh_brg_cfg_state val,
```
This is for all these functions in the client. I know the names will be a little bit long, but is it not better to keep the names closer to what is in the spec. The spec name is subnet_bridge_set, and then it is easier to connect this to the spec function. I think I would prefer to only change the brg_cfg.h names.

HaavardRei replied:
There is some disagreement on this front, thoughts @PavelVPV and @alxelax?
I agree that it would be easier to connect it to the spec with the longer names, but in the other model APIs we have a tendency to go for the shorter names.

PavelVPV replied:
I tend to think consistency with our other models API prevail over consistency with spec terminology simply because that makes it easier to work with the codebase. I'd probably do that as Ingar suggested from the beginning, but the pattern is already defined in other API, so my suggestion to follow it.

alxelax replied:
I agree with @PavelVPV. I still think that the double mentioning `bridge` in one form or another doesn't add any value.

Balaklaka replied:
Ok, to keep consistency with rest of the code base is more important.

HaavardRei commented on include/zephyr/bluetooth/mesh/brg_cfg.h:
```diff
@@ -21,20 +21,20 @@ extern "C" {
 
 /** Subnet Bridge states */
-enum bt_mesh_subnet_bridge_state {
+enum bt_mesh_brg_cfg_state {
```
The same argument could be made for all the functions whose names are cut/shortened. I think the comment above (Subnet Bridge states) is sufficient to lead you to 4.2.41 (Searching for _Subnet bridge state_ gives 4.2.41 as the first result), but I can add a reference to the spec in the comment if you insist!

Balaklaka replied:
Because we are cutting the names in the header files. We can do that here too.

alxelax reviewed (commented):
LGTM, just fix bsim and others tests.

HaavardRei committed:
Bluetooth: Mesh: Remove duplicate brg declaration

Removes duplicate declaration of the bridging direction, keeping the
one in the public header file.

Signed-off-by: Håvard Reierstad <haavard.reierstad@nordicsemi.no>

Balaklaka reviewed (commented):
We  need to fix that the BT_MESH_BRG_CFG_DIR_ONEWAY/BT_MESH_BRG_CFG_DIR_TWOWAY are defined twice.
